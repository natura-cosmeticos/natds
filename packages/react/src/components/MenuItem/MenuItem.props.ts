import React from 'react'

export interface MenuItemProps {
  /**
   * Specify an optional className to be added to your MenuItem
   */
  children: React.ReactNode

  /**
   * Specify an optional className to be added to your MenuItem
   */
  className?: string

  /**
   * Click handler
   */
  onClick?: (e?: React.MouseEvent<HTMLLIElement>) => void

  /**
   * Optional ID for testing
   */
  testID?: string;

  /**
   * Specify whether the MenuItem should be disabled, or not
   * @default true
   */
  disabled?: boolean;

  /**
   * Used to show for the user the current selected menu.
   * If `true`, the primary color will be applied on background
   * @default false
   */
  selected?: boolean;

  /**
   * Used to show for the user the parent menu activated.
   * If `true`, a low emphasis color will be applied on background
   * @default false
   */
  actived?: boolean;

  /**
   * If `true`, a border will be displayed on the left of the container.
   * @default false
   */
  submenu?: boolean;

  /**
   *
   */
  href?: React.AnchorHTMLAttributes<HTMLAnchorElement>['href']

  /**
   *
   */
  hrefLang?: React.AnchorHTMLAttributes<HTMLAnchorElement>['hrefLang']
}
