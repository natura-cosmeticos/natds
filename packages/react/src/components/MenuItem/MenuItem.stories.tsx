import React from 'react'
import { Story, Meta } from '@storybook/react'
import { MenuItem, MenuItemProps } from '.'

const componentStatus = `
---

**NOTE FOR UXs**: This component is available in the following variants:

  - ❌ Standard

With the following attribute statuses:

  - ✅ **Submenu**
  - ❌ **Actived**
  - ❌ **Selected**
  - ❌ **Disabled**
  - ✅ **Add** (available but with alternative composition)
  - **Action**
      - ✅ \`left\` (available but with alternative composition)
      - ✅ \`right\` (available but with alternative composition)

---
`

export default {
  title: 'Components/MenuItem',
  component: MenuItem,
  parameters: {
    componentSubtitle: '',
    docs: { description: { component: componentStatus } }
  }
} as Meta

export const Playground: Story<MenuItemProps> = (args) => (
  <ul style={{ padding: 0, margin: 0 }}>
    <MenuItem {...args} />
  </ul>
)
Playground.args = {
  children: 'Menu Item'
}

export const Submenu: Story<MenuItemProps> = Playground.bind({})
Submenu.args = {
  ...Playground.args,
  submenu: true
}

export const Disabled: Story<MenuItemProps> = Playground.bind({})
Disabled.args = {
  ...Playground.args,
  disabled: true
}
