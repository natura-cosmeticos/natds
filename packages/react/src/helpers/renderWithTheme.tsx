import React from 'react'
import { JssProvider, SheetsRegistry } from 'react-jss'
import { render } from '@testing-library/react'
import { ThemeProvider } from '../ThemeProvider'

const renderWithTheme = (component) => {
  const sheets = new SheetsRegistry()

  const renderTree = render(
    <JssProvider registry={sheets}>
      <ThemeProvider>
        {component}
      </ThemeProvider>
    </JssProvider>
  )

  return { styles: sheets.toString(), component: renderTree }
}

export default renderWithTheme
