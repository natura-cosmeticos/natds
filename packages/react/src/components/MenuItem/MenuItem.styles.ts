/* eslint-disable complexity */
import { createUseStyles } from 'react-jss'
import { Theme } from '@naturacosmeticos/natds-themes'
import { MenuItemProps } from './MenuItem.props'

const styles = createUseStyles((theme: Theme) => ({
  menuItem: ({ disabled = false, submenu = false }: MenuItemProps) => ({
    alignItems: 'center',
    borderLeft: submenu ? `1px solid ${theme.color.lowEmphasis}` : 'none',
    color: disabled ? theme.color.lowEmphasis : theme.color.highEmphasis,
    cursor: disabled ? 'default' : 'pointer',
    display: 'flex',
    fontFamily: [
      theme.navigationDrawer.section.item.primary.fontFamily,
      theme.navigationDrawer.section.item.fallback.fontFamily
    ],
    fontSize: theme.body1.fontSize,
    fontWeight: theme.body1.fontWeight,
    height: theme.size.semiX,
    letterSpacing: theme.body1.letterSpacing,
    lineHeight: theme.body1.lineHeight,
    listStyle: 'none',
    padding: [0, theme.spacing.small],
    width: '100%',
    '&:hover': {
      backgroundColor: disabled ? 'transparent' : theme.color.neutral100
    },
    '&:focus': {
      backgroundColor: disabled ? 'transparent' : theme.color.neutral200
    }
  })
}))

export default styles
