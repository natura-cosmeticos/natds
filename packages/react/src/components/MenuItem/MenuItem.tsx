import React from 'react'
import styles from './MenuItem.styles'
import { MenuItemProps } from './MenuItem.props'

const MenuItem = React.forwardRef<HTMLLIElement, MenuItemProps>(({
  className = '',
  testID,
  onClick,
  children,
  href,
  ...props
}, ref) => {
  const { menuItem } = styles(props)

  return (
    <li
      data-testid={testID}
      className={`${className} ${menuItem}`}
      ref={ref}
      tabIndex={0}
      onClick={onClick}
    >
      <a href={href} {...props}>
        {children}
      </a>
    </li>
  )
})

export default MenuItem
