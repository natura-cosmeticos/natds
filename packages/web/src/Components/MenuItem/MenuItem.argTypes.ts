import { ArgType } from '@storybook/addons'
import { IMenuItemProps } from './MenuItem.props'

export const argTypes : Record<keyof IMenuItemProps | string, ArgType> = {
  children: {
    description: 'Menu item contents.',
    table: {
      type: { summary: 'node' }
    }
  },
  classes: {
    control: { type: 'object' },
    description: `
Override or extend the styles applied to the component.

See the [CSS API](https://material-ui.com/api/menu-item/#css) for more details.
`,
    table: {
      type: { summary: 'object' }
    },
    type: { name: 'object', required: false }
  },
  component: {
    description: 'The component used for the root node. Either a string to use a DOM element or a component.',
    table: {
      defaultValue: { summary: '"li"' },
      type: { summary: 'elementType' }
    }
  },
  dense: {
    control: { type: 'boolean' },
    description: 'If true, compact vertical padding designed for keyboard and mouse input will be used.',
    table: {
      type: { summary: 'bool' }
    },
    type: { name: 'boolean', required: false }
  },
  disableGutters: {
    control: { type: 'boolean' },
    description: 'If `true`, the left and right padding is removed.',
    table: {
      defaultValue: { summary: false },
      type: { summary: 'bool' }
    },
    type: { name: 'boolean', required: false }
  }
}
export default argTypes
