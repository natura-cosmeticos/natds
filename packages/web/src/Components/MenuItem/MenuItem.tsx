import * as React from 'react'
import MaterialMenuItem from '@material-ui/core/MenuItem'
import { IMenuItemProps } from './MenuItem.props'

export { IMenuItemProps } from './MenuItem.props'

/**
 * ## Importing
 *
 * ```
 * import { MenuItem } from '@naturacosmeticos/natds-web';
 * ```
 */
export const MenuItem = React.forwardRef<HTMLLIElement, IMenuItemProps>(
  (props: IMenuItemProps, ref) => {
    const { button, ...otherProps } = props

    return (
      <MaterialMenuItem
        {...otherProps}
        button={button as true | undefined}
        ref={ref}
      />
    )
  }
)
MenuItem.displayName = 'MenuItem'
export default MenuItem
