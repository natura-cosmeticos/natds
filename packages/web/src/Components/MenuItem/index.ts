import MenuItem from './MenuItem'

export { IMenuItemProps } from './MenuItem.props'
export { MenuItem }
export default MenuItem
