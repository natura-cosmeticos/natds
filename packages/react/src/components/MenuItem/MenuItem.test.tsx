import React from 'react'
import MenuItem from '.'
import renderWithTheme from '../../helpers/renderWithTheme'

describe('MenuItem component', () => {
  it('should render correctly with default props', () => {
    const { styles, component } = renderWithTheme(<MenuItem>Menu Item</MenuItem>)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
})
