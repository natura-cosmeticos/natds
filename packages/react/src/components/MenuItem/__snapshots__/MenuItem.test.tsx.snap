// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem component should render correctly with default props 1`] = `
Array [
  ".menuItem-0-2-1 {}
.menuItem-d0-0-2-2 {
  color: #333333;
  width: 100%;
  cursor: pointer;
  height: 40px;
  display: flex;
  padding: 0 16px;
  font-size: 16px;
  list-style: none;
  align-items: center;
  border-left: none;
  font-family: Roboto, Arial;
  font-weight: 400;
  line-height: 1.5;
  letter-spacing: 0.51px;
}
.menuItem-d0-0-2-2:hover {
  background-color: #f1f3f5;
}
.menuItem-d0-0-2-2:focus {
  background-color: #e9ecef;
}",
  <div>
    <li
      class=" menuItem-0-2-1 menuItem-d0-0-2-2"
      tabindex="0"
    >
      <a>
        Menu Item
      </a>
    </li>
  </div>,
]
`;
